Mathlib/RepresentationTheory/Induced.lean

```diff
@@ -0,0 +1,158 @@
+`(k[H] ⊗[k] A)_G` with the `G`-representation on `k[H]` defined by `φ`. -/
+abbrev IndV := Coinvariants (V := TensorProduct k (H →₀ k) A)
+  (Representation.tprod ((leftRegular k H).comp φ) ρ)
+
+/-- Given a group homomorphism `φ : G →* H` and a `G`-representation `(A, ρ)`, this is the
+`H → A →ₗ[k] (k[H] ⊗[k] A)_G` sending `h, a` to `⟦h ⊗ₜ a⟧`. -/
+noncomputable abbrev IndV.mk (h : H) : A →ₗ[k] IndV φ ρ :=
+  Coinvariants.mk _ ∘ₗ TensorProduct.mk k _ _ (single h 1)
+
+@[ext]
+lemma IndV.hom_ext {f g : IndV φ ρ →ₗ[k] B}
+    (hfg : ∀ h : H, f ∘ₗ IndV.mk φ ρ h = g ∘ₗ IndV.mk φ ρ h) : f = g :=
+  Coinvariants.hom_ext <| TensorProduct.ext <| Finsupp.lhom_ext' fun h =>
+    LinearMap.ext_ring <| hfg h
+
+/-- Given a group homomorphism `φ : G →* H` and a `G`-representation `A`, this is
+`(k[H] ⊗[k] A)_G` equipped with the `H`-representation defined by sending `h : H` and `⟦h₁ ⊗ₜ a⟧`
+to `⟦h₁h⁻¹ ⊗ₜ a⟧`. -/
+@[simps]
+noncomputable def ind : Representation k H (IndV φ ρ) where
+  toFun h := Coinvariants.map _ _ ((lmapDomain k k fun x => x * h⁻¹).rTensor _)
+    fun _ => by ext; simp [mul_assoc]
+  map_one' := by ext; simp [IndV, IndV.mk]
+  map_mul' _ _ := by ext; simp [IndV, IndV.mk, mul_assoc]
+
+lemma ind_mk (h₁ h₂ : H) (a : A) :
+    ind φ ρ h₁ (IndV.mk _ _ h₂ a) = IndV.mk _ _ (h₂ * h₁⁻¹) a := by
+  simp
+
+end Representation
+
+namespace Rep
+
+open CategoryTheory Finsupp
+
+variable {k G H : Type u} [CommRing k] [Group G] [Group H] (φ : G →* H) (A : Rep k G)
+
+section Ind
+
+/-- Given a group homomorphism `φ : G →* H` and a `G`-representation `A`, this is
+`(k[H] ⊗[k] A)_G` equipped with the `H`-representation defined by sending `h : H` and `⟦h₁ ⊗ₜ a⟧`
+to `⟦h₁h⁻¹ ⊗ₜ a⟧`. -/
+noncomputable abbrev ind : Rep k H := Rep.of (A.ρ.ind φ)
+
+/-- Given a group homomorphism `φ : G →* H`, a morphism of `G`-representations `f : A ⟶ B` induces
+a morphism of `H`-representations `(k[H] ⊗[k] A)_G ⟶ (k[H] ⊗[k] B)_G`. -/
+@[simps]
+noncomputable def indMap {A B : Rep k G} (f : A ⟶ B) : ind φ A ⟶ ind φ B where
+  hom := ModuleCat.ofHom <| Representation.Coinvariants.map _ _
+    (LinearMap.lTensor (H →₀ k) f.hom.hom) fun g => by ext; simp [hom_comm_apply]
+  comm _ := by
+    ext
+    simp [ModuleCat.endRingEquiv]
+
+variable (k) in
+/-- Given a group homomorphism `φ : G →* H`, this is the functor sending a `G`-representation `A`
+to the induced `H`-representation `ind φ A`, with action on maps induced by left tensoring. -/
+@[simps obj map]
+noncomputable def indFunctor : Rep k G ⥤ Rep k H where
+  obj A := ind φ A
+  map f := indMap φ f
+  map_id _ := by ext; rfl
+  map_comp _ _ := by ext; rfl
+
+end Ind
+section Adjunction
+
+open Representation
+
+variable (B : Rep k H)
+
+/-- Given a group homomorphism `φ : G →* H`, an `H`-representation `B`, and a `G`-representation
+`A`, there is a `k`-linear equivalence between the `H`-representation morphisms `ind φ A ⟶ B` and
+the `G`-representation morphisms `A ⟶ B`. -/
+@[simps]
+noncomputable def indResHomLEquiv :
```

Comment:
I'm inclined to drop the `L`. I think that improves the readability of the name.
```suggestion
noncomputable def indResHomEquiv :
```